src/gui/widgets/main_window.py

```diff
@@ -366,60 +370,83 @@ def updateCompositions(self):
         self.mainWidget.sampleCompositionTable.farmCompositions()
         self.mainWidget.containerCompositionTable.farmCompositions()
 
-    def updatePlots(self):
+    def updateResults(self):
         if self.mainWidget.objectStack.currentIndex() == 4:
+            sample = self.mainWidget.objectTree.currentObject()
             self.mainWidget.sampleChart = GudPyChart(
                 self.gudrunFile.instrument.dataFileType,
                 self.gudrunFile.instrument.GudrunInputFileDir,
-                sample=self.mainWidget.objectTree.currentObject()
+                sample=sample
             )
             self.mainWidget.sampleStructureFactorChartView.setChart(
                 self.mainWidget.sampleChart
             )
             self.mainWidget.sampleRDFChart = GudPyChart(
                 self.gudrunFile.instrument.dataFileType,
                 self.gudrunFile.instrument.GudrunInputFileDir,
-                sample=self.mainWidget.objectTree.currentObject(),
+                sample=sample,
                 plotMode=PlotModes.RADIAL_DISTRIBUTION_FUNCTIONS
             )
             self.mainWidget.sampleRDFChartView.setChart(
                 self.mainWidget.sampleRDFChart
             )
-        else:
-            self.mainWidget.allSamplesChart = GudPyChart(
-                self.gudrunFile.instrument.dataFileType,
-                self.gudrunFile.instrument.GudrunInputFileDir,
-                samples=[
-                    sample
-                    for sampleBackground in self.gudrunFile.sampleBackgrounds
-                    for sample in sampleBackground.samples
-                ]
-            )
-            self.mainWidget.allSampleStructureFactorChartView.setChart(
-                self.mainWidget.allSamplesChart
+            path = sample.dataFiles.dataFiles[0].replace(
+                self.gudrunFile.instrument.dataFileType, "gud"
             )
+            if not os.path.exists(path):
+                path = os.path.join(
+                    self.gudrunFile.instrument.GudrunInputFileDir, path
+                )
+            if os.path.exists(path):
+                gf = GudFile(path)
+                dcsLevel = gf.averageLevelMergedDCS
+                self.mainWidget.dcsLabel.setText(
+                    f"Average level of merged DCS: {dcsLevel}"
```

Comment:
Just a small suggestion to reduce text and simplify:
```suggestion
                    f"DCS Level: {dcsLevel}"
```